File: apis/src/main/java/com/beat/apis/booking/application/dto/GuestBookingResponse.java
package com.beat.apis.booking.application.dto;

import com.beat.apis.booking.application.dto.BookingStatusType;
import com.beat.apis.performance.application.dto.BankNameType;
import com.beat.apis.schedule.application.dto.ScheduleNumberType;

import java.time.LocalDateTime;

public record GuestBookingResponse(
	Long bookingId,
	Long scheduleId,
	Long userId,
	int purchaseTicketCount,
	ScheduleNumberType scheduleNumber,
	String bookerName,
	String bookerPhoneNumber,
	BookingStatusType bookingStatus,
	BankNameType bankName,
	String accountNumber,
	int totalPaymentAmount,
	LocalDateTime createdAt
) {
	public static GuestBookingResponse of(
		Long bookingId,
		Long scheduleId,
		Long userId,
		int purchaseTicketCount,
		ScheduleNumberType scheduleNumber,
		String bookerName,
		String bookerPhoneNumber,
		BookingStatusType bookingStatus,
		BankNameType bankName,
		String accountNumber,
		int totalPaymentAmount,
		LocalDateTime createdAt) {
		return new GuestBookingResponse(
			bookingId,
			scheduleId,
			userId,
			purchaseTicketCount,
			scheduleNumber,
			bookerName,
			bookerPhoneNumber,
			bookingStatus,
			bankName,
			accountNumber,
			totalPaymentAmount,
			createdAt);
	}
}

File: apis/src/main/kotlin/com/beat/apis/booking/api/response/GuestBookingResponse.kt
package com.beat.apis.booking.api.response

import com.beat.apis.booking.api.type.BookingStatusType
import com.beat.apis.booking.application.result.BookingCreationResult
import com.beat.apis.performance.api.type.BankNameType
import com.beat.apis.schedule.api.type.ScheduleNumberType
import java.time.LocalDateTime

@ConsistentCopyVisibility
data class GuestBookingResponse private constructor(
    val bookingId: Long?,
    val scheduleId: Long?,
    val userId: Long?,
    val purchaseTicketCount: Int,
    val scheduleNumber: ScheduleNumberType?,
    val bookerName: String?,
    val bookerPhoneNumber: String?,
    val bookingStatus: BookingStatusType?,
    val bankName: BankNameType?,
    val accountNumber: String?,
    val totalPaymentAmount: Int,
    val createdAt: LocalDateTime?,
) {
    companion object {
        fun from(result: BookingCreationResult): GuestBookingResponse = GuestBookingResponse(
            bookingId = result.bookingId,
            scheduleId = result.scheduleId,
            userId = result.userId,
            purchaseTicketCount = result.purchaseTicketCount,
            scheduleNumber = result.scheduleNumber?.let(ScheduleNumberType::valueOf),
            bookerName = result.bookerName,
            bookerPhoneNumber = result.bookerPhoneNumber,
            bookingStatus = result.bookingStatus?.let(BookingStatusType::valueOf),
            bankName = result.bankName?.let(BankNameType::valueOf),
            accountNumber = result.accountNumber,
            totalPaymentAmount = result.totalPaymentAmount,
            createdAt = result.createdAt,
        )
    }
}
